blog/2024-03-27-reductstore-vs-timescale/index.md

```diff
@@ -0,0 +1,220 @@
+
+```python
+if __name__ == "__main__":
+    print(f"Chunk size={BLOB_SIZE / 1000_000} Mb, count={BLOB_COUNT}")
+    ts = time.time()
+    size = write_to_timescale()
+    print(f"Write {size / 1000_000} Mb to TimescaleDB: {time.time() - ts} s")
+
+    ts_read = time.time()
+    size = read_from_timescale(ts, time.time())
+    print(f"Read {size / 1000_000} Mb from TimescaleDB: {time.time() - ts_read} s")
+
+    loop = asyncio.new_event_loop()
+    ts = time.time()
+    size = loop.run_until_complete(write_to_reduct())
+    print(f"Write {size / 1000_000} Mb to ReductStore: {time.time() - ts} s")
+
+    ts_read = time.time()
+    size = loop.run_until_complete(read_from_reduct(ts, time.time()))
+    print(f"Read {size / 1000_000} Mb from ReductStore: {time.time() - ts_read} s")
+```
+
+For testing purposes, we need to run the databases. This can easily be done using docker-compose:
+
+```yaml
+version: "3"
+services:
+  timescale:
+    image: timescale/timescaledb:latest-pg14
+    ports:
+      - "5432:5432"
+    environment:
+      POSTGRES_USER: postgres
+      POSTGRES_PASSWORD: postgres
+    volumes:
+      - ${PWD}/data/timescale:/var/lib/postgresql/data
+
+  reductstore:
+    image: reduct/store:latest
+    ports:
+      - "8383:8383"
+    environment:
+      RS_API_TOKEN: reductstore
+      RS_BUCKET_1_NAME: benchmark
+      RS_BUCKET_1_QUTA_TYPE: FIFO
+      RS_BUCKET_1_QUOTA_SIZE: 50TB
+
+    volumes:
+      - ${PWD}/data/reductstore:/data
+```
+
+Take note, ReductStore enables the provisioning of any resources with environment variables. In this case, we create a bucket called `benchmark` with a 50TB quota. Your DevOps team will appreciate it.
+
+Let's set up and run benchmarks:
+
+```yaml
+docker-compose up -d
+python3 main.py
+```
+
+## Results
+
+The script displays results for the specified `BLOB_SIZE` and `SIZE_COUNT`. On my device, which has an NVMe drive, here are the results I obtained:
+
+| Chunk Size | Operation | TimescaleDB, blob/s | ReductStore, blob/s |
+| --- | --- | --- | --- |
+| 10 KB | Write | 1557 | 1500 |
+|  | Read | 1333 | 1280 |
+| 100 KB | Write | 447 | 1366 |
+|  | Read | 353 | 1120 |
+| 1 MB | Write | 53 | 571 |
+|  | Read | 40 | 382 |
+| 10 Mb | Write | 5 | 70 |
+|  | Read | 4 | 38 |
+
+Based on the benchmark results, ReductStore and TimescaleDB demonstrate similar performance with blobs 10KB and smaller. However, for larger blobs, ReductStore significantly outperforms TimescaleDB in both write and read operations.
+
+## Conclusions
+
+TimescaleDB excels as a time series database and can be used to store small blobs. It's a good choice if you need to store structured data and want to avoid additional storage for blob data. However, if you have blobs larger than 10KB, ReductStore might be a better option. It offers better performance and provides a retention policy based on disk usage, which is crucial when the amount of data stored over time is unpredictable.
```

Comment:
The "retention policy based on disk usage" comes out only in the conclusion. Could you add a chapter before that tells "Other Considerations" ?